### PR TITLE
conversation and conversation list rendering

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: go/protocol/chat1/extras.go [hunk 1]
```diff
@@ -21,3 +21,35 @@ func (id TLFID) EqString(other fmt.Stringer) bool {
 func (cid ConversationID) String() string {
 	return strconv.FormatUint(uint64(cid), 10)
 }
+
+func (t MessageType) String() string {
```
Comment (maxtaco) [inline go/protocol/chat1/extras.go]: i wonder if we should make the avdl compiler output this?

File: go/client/chat_cli_rendering.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		for ; iConversation < len(v) && len(v[iConversation].Messages) == 0; iConversation++ {
+		}
+		if iConversation == len(v) {
+			return nil
+		}
+		ret := []string{
+			fmt.Sprintf("[%d]", iConversation),
+			v.renderConversationName(20, iConversation),
+			messageFormatter(v[iConversation].Messages[0]).renderAuthorAndTime(),
+			messageFormatter(v[iConversation].Messages[0]).renderMessage(40),
```
Comment (maxtaco) [inline go/client/chat_cli_rendering.go]: where does the 40 come from?

We can get the terminal size in the `Terminal` class. I would expose it to the `TerminalUI` interface and then call it here..

File: go/client/chat_cli_rendering.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	iConversation := 0
+	ui.TablifyAlignRight(nil, func() []string {
+		for ; iConversation < len(v) && len(v[iConversation].Messages) == 0; iConversation++ {
+		}
+		if iConversation == len(v) {
+			return nil
+		}
+		ret := []string{
+			fmt.Sprintf("[%d]", iConversation),
+			v.renderConversationName(20, iConversation),
```
Comment (maxtaco) [inline go/client/chat_cli_rendering.go]: similarly, where does the `20` come from?

File: go/protocol/chat1/extras.go (hunk 1 above)
Comment (songgao) [inline go/protocol/chat1/extras.go]: For all enum types? That'd be great!

File: go/client/chat_cli_rendering.go (hunk 2 above)
Comment (songgao) [inline go/client/chat_cli_rendering.go]: Sounds good. Will do!

**Comment by maxtaco** [comment]
👍

File: go/protocol/chat1/extras.go (hunk 1 above)
Comment (maxtaco) [inline go/protocol/chat1/extras.go]: Cool I saw the JIRA ticket...

**Comment by songgao** [comment]
Dear GitHub,

Please close https://github.com/keybase/client/pull/4073 when this one's merged.

Yours Truly,
Song Gao

**Comment by songgao** [comment]
Oops doesn't work from comments? hmm..

**Pull request merged**